icepyx/core/orders.py

```diff
@@ -0,0 +1,203 @@
+            The response from the Harmony API if the order is resumed, otherwise None.
+        """
+        if self.type == "subset":
+            return self.harmony_api.resume_order(self.job_id)
+        return None
+
+    def skip_preview(self) -> Union[Dict[str, Any], None]:
+        """
+        Resume the order if it has been paused due to exceeding granule limits.
+
+        Returns
+        -------
+        dict or None
+            The response from the Harmony API if the order is resumed, otherwise None.
+        """
+        if self.type == "subset":
+            return self.harmony_api.skip_preview(self.job_id)
+        return None
+
+    def pause(self) -> Union[Dict[str, Any], None]:
+        """
+        Pause the order.
+
+        Returns
+        -------
+        dict or None
+            The response from the Harmony API if the order is paused, otherwise None.
+        """
+        if self.type == "subset":
+            return self.harmony_api.pause_order(self.job_id)
+        return None
+
+    def status(self) -> Dict[str, Any]:
+        """
+        Retrieve the status of the order.
+
+        Returns
+        -------
+        dict
+            A dictionary containing the order status and related metadata.
+        """
+        if self.type == "subset":
+            status = self.harmony_api.check_order_status(self.job_id)
+            # so users don't accidentally order again
+            status.pop("request")
+            status["order_url"] = self.HARMONY_BASE_URL + str(self.job_id)
+            return status
+        return {"status": "complete"}
+
+    def download_granules(self, path, overwrite=False) -> Union[list, None]:
+        """
+        Download the granules for the order.
+
+        Parameters
+        ----------
+        path : str or Path
+            The directory where granules should be saved.
+        overwrite : bool, optional
+            Whether to overwrite existing files (default is False).
+        Returns
+        -------
+        list or None
+            A list of downloaded granules
+
+        """
+        return self.download(path, overwrite=overwrite)
+
+    def download(self, path, overwrite=False) -> Union[list, None]:
+        """
+        Download the granules for the order, blocking until they are ready if necessary.
+
+        Parameters
+        ----------
+        path : str or Path
+            The directory where granules should be saved.
+        overwrite : bool, optional
+            Whether to overwrite existing files (default is False).
+
+        Returns
+        -------
+        list or None
+            A list of downloaded granules
+        """
+        path = Path(path)
+        path.mkdir(parents=True, exist_ok=True)
+        if self.type == "subset":
+            return self.harmony_api.download_granules(
+                download_dir=str(path), overwrite=overwrite
+            )
+        else:
+            return earthaccess.download(self.granules, local_path=path)
```

Comment:
aren't these (`download_granules` and `download`) effectively duplicate functions?

Reply:
they are equivalent, I think of them as aliases because the name of the function in the Query class is `download_granules` so if I user wants to map that to this class it would work the same.

Reply:
Currently the user interacts with the icepyx Query class; the future goal is that they will use an earthaccess Results class object (that's currently being created). In both cases, they would not interact directly with the Orders object, so  what is the use case for an `order.download_granules` alias versus mapping `query.download_granules` to `order.download`?

Reply:
I think `earthaccess` will use an order's class as well, because it's more intuitive to control orders on its own abstraction. In this case I was switching between the Query class and the DataOrder, and noticed that I had named `download()` in the order, my preference would be that Query would adopt that but it would be another breaking change.

Reply:
> because it's more intuitive to control orders on its own abstraction

Agreed!

> noticed that I had named download() in the order, my preference would be that Query would adopt that but it would be another breaking change.

Thanks for this explanation - that totally makes sense. Would this be a good case for a deprecation warning? I agree it would be nice if we could not break this fundamental, but if we're pretty certain that `download()` will be the future function name that aligns with earthaccess/plugin architecture, should we use it as an opportunity to suggest users update their code for that eventuality?